tauri-apps/tauri-discord-bot
# Feat/reaction roles

- Collapse `HELP_CHANNELS` and `AUTO_THREAD_CHANNELS` into a single `AUTO_THREAD_CHANNELS` const
- Move `thread.ts` up a directory
- Rename `TEST_GUILD_ID` to `GUILD_ID`
- Add `REACTION_ROLE` and `REACTION_ROLE_CHANNEL` as constants for the Reaction Role logic
- Add `messageReactionAdd.ts` and `messageReactionRemove.ts` events
- Add `utils/reactionHandler.ts` that will check if a reaction should be handled and will send a message explaining the reaction roles
- Add message on startup to explain reaction roles
- Add intents for `GUILD_MESSAGE_REACTIONS` (to read reactions) and `GUILD_MEMBERS` (to loop through members and add a role for those who have reacted but don't have the role such as if the bot was down)
- Remove unused `utils/autocomplete.ts` file
- Fix issue where `::` was being recognised as a URL but wasn't able to be parsed, falls back to the passed string if that's the case
